code/modules/surgery/implant.dm

```diff
@@ -1,228 +1,284 @@
-		if (!ishuman(target))
-			return 0
-		var/obj/item/organ/external/affected = target.get_organ(target_zone)
-		var/can_fit = affected && !affected.hidden && affected.cavity && tool.w_class <= get_max_wclass(affected)
-		return ..() && can_fit
 
-	begin_step(mob/user, mob/living/carbon/human/target, target_zone, obj/item/tool)
-		var/obj/item/organ/external/affected = target.get_organ(target_zone)
+/datum/surgery_step/cavity/place_item/can_use(mob/living/user, mob/living/carbon/human/target, target_zone, obj/item/tool,datum/surgery/surgery)
+	if (!ishuman(target))
+		return 0
+	var/obj/item/organ/external/affected = target.get_organ(target_zone)
+	var/can_fit = affected && !affected.hidden && affected.cavity && tool.w_class <= get_max_wclass(affected)
+	return ..() && can_fit
+
+/datum/surgery_step/cavity/place_item/begin_step(mob/living/user, mob/living/carbon/human/target, target_zone, obj/item/tool,datum/surgery/surgery)
+	var/obj/item/organ/external/affected = target.get_organ(target_zone)
+	for(var/obj/item/I in target.internal_organs)
+		if(!istype(I, /obj/item/organ))
+			IC = I
+			break
+	if(tool)
 		user.visible_message("[user] starts putting \the [tool] inside [target]'s [get_cavity(affected)] cavity.", \
 		"You start putting \the [tool] inside [target]'s [get_cavity(affected)] cavity." )
-		target.custom_pain("The pain in your chest is living hell!",1)
-		..()
-
-	end_step(mob/living/user, mob/living/carbon/human/target, target_zone, obj/item/tool)
-		var/obj/item/organ/external/chest/affected = target.get_organ(target_zone)
-
-		if(istype(tool, /obj/item/weapon/disk/nuclear))
-			user << "Central command would kill you if you implanted the disk into someone."
+	else
+		user.visible_message("[user] checks for items in [target]'s [target_zone].", "<span class='notice'>You check for items in [target]'s [target_zone]...</span>")
+	target.custom_pain("The pain in your chest is living hell!",1)
+	..()
+
+/datum/surgery_step/cavity/place_item/end_step(mob/living/user, mob/living/carbon/human/target, target_zone, obj/item/tool,datum/surgery/surgery)
+	var/obj/item/organ/external/chest/affected = target.get_organ(target_zone)
+
+	if(istype(tool, /obj/item/weapon/disk/nuclear))
+		user << "Central command would kill you if you implanted the disk into someone."
+		return 0//fail
+
+	if(tool)
+		user.visible_message("\blue [user] puts \the [tool] inside [target]'s [get_cavity(affected)] cavity.", \
+		"\blue You put \the [tool] inside [target]'s [get_cavity(affected)] cavity." )
+		if (IC || (tool.w_class > get_max_wclass(affected)/2 && prob(50) && !(affected.status & ORGAN_ROBOT)))
+			user << "\red You tear some vessels trying to fit the object in the cavity."
+			var/datum/wound/internal_bleeding/I = new ()
+			affected.wounds += I
+			affected.owner.custom_pain("You feel something rip in your [affected.name]!", 1)
+		user.drop_item()
+		affected.hidden = tool
+		target.internal_organs += tool
```

Comment:
also its not calling the insert function for an organ, its just adding it to the 'things that we have in our body' list..though i CAN double check.